stwcs/updatewcs/__init__.py

```diff
@@ -30,7 +30,8 @@
 warnings.filterwarnings("ignore", message="^Some non-standard WCS keywords were excluded:", module="astropy.wcs")
 
 def updatewcs(input, vacorr=True, tddcorr=True, npolcorr=True, d2imcorr=True,
-              checkfiles=True, verbose=False, use_db=True, all_wcs=False):
+              checkfiles=True, verbose=False, use_db=True,
+              all_wcs=False, remove_duplicates=True):
```

Comment:
Actually, is there any instance when one would want to set `remove_duplicates=False`? I have a feeling that the answer would be "NOT" and then this parameter should not even be exposed to the user - just fix the code to take care of the duplicates. Maybe add a log entry if duplicates are detected.

Reply:
This gets to the heart of how transparent should the code be, since this code can be run on data potentially already modified by the end user.  The user may have already tweaked the alignment for a few different catalogs but given all the new solutions the same `HDRNAME` value.  This was actually pretty easy to do in the past and I am not going to assume we have eliminated the possibility now.  Should this code simply come along, delete all their work while telling them all about it in the process through prominent log messages without giving them the ability to save those solutions instead?  That would not result in happy users!  Also, since this code is modifying the input file in many documented manners already, we should allow the user to control this additional new and documented operation.  This parameter also provides the most obvious way to document this operation since it stands out as a separate entry in the documentation.  In short, I see no harm in adding this parameter, and do not want to risk the very real possibility of causing the end user more problems than it is solving by doing something they have no control over.

Reply:
But what will happen if users do not know that they have duplicate headerlets (just as do not, most likely). How are they going to know that they need to remove duplicates? Will they just run `updatewcs` with the default values, like most users do? Won't this result in unhappy users too?

Reply:
The documentation makes it clear that it will, by default, delete all duplicates due to the presence of this parameter.  The updated docstring also adds a discussion as to what this means for them (it may cause an Exception).  Having this exposed at least gives them notice as to this behavior, even if they only realize it on their own after the fact.  It is always better to at least expose the option rather than hide it and do it under the covers so to speak.  The default behavior will at least insure that the final output will not have duplicates and will work with the rest of the headerlet functions, unlike what can happen if duplicates are left in place as demonstrated by the regression test that failed and spurred this PR in the first place.